### PR TITLE
coreos-overlay: Remove unused sys-fs/fuse package

### PR DESCRIPTION
**Pull request by glevand**
This is a re-submission of the two patches from PR https://github.com/coreos/coreos-overlay/pull/1288 which remove the sys-fs/fuse package, along with a needed patch to open-vm-tools which adds the missing --without-fuse configure option.

I tested this series on a clean system build with this: './chromite/bin/cros_sdk --nousepkg --replace -- ~/trunk/src/scripts/build_packages --board amd64-usr --nousepkg', and the build was successful.

I'll prepare another PR to remove the unused fuse related packages from portage-stable, but will wait to verify all goes well with this series before merging the portage-stable patches.

### CODE REVIEW

File: app-emulation/open-vm-tools/open-vm-tools-9.10.0.ebuild [hunk 1]
```diff
@@ -35,6 +35,10 @@ RDEPEND="dnet? ( dev-libs/libdnet )
 
 S="${WORKDIR}/${PN}"
 
+PATCHES=(
```
Comment (marineam) [inline app-emulation/open-vm-tools/open-vm-tools-9.10.0.ebuild]: please add a -r1 to the ebuild version.

File: coreos-base/coreos-dev/coreos-dev-0.1.0.ebuild [hunk 2]
```diff
@@ -44,9 +44,7 @@ RDEPEND="
 	sys-apps/which
 	sys-devel/gcc
 	sys-devel/gdb
-	sys-fs/fuse
```
Comment (marineam) [inline coreos-base/coreos-dev/coreos-dev-0.1.0.ebuild]: please bump the -r revision in the symlink too

File: coreos-base/hard-host-depends/hard-host-depends-0.0.1.ebuild [hunk 3]
```diff
@@ -78,7 +78,6 @@ RDEPEND="${RDEPEND}
 	sys-apps/systemd
 	!sys-apps/nih-dbus-tool
 	=sys-devel/automake-1.10*
-	sys-fs/sshfs-fuse
```
Comment (marineam) [inline coreos-base/hard-host-depends/hard-host-depends-0.0.1.ebuild]: ditto, need to update revision

**Comment by marineam** [comment]
looks good, just missing ebuild revision updates :)

**Comment by glevand** [comment]
I remember updating those revisions last time, they must have gone missing along the way...

**Comment by marineam** [comment]
LGTM

**Pull request merged**